monai/transforms/compose.py

```diff
@@ -334,6 +334,18 @@ def __call__(self, input_, start=0, end=None, threading=False, lazy: bool | None
 
         return result
 
+    def is_random(self):
```

Comment:
Thanks for the PR!
Do we have another special reason to add this `is_random` function?
Looks like we could directly use this `flatten` function in the `CacheDataset`? Do you have concerns about this easy solution?
https://github.com/Project-MONAI/MONAI/blob/8a70678baa976a01274f55837e339f2c2975dec7/monai/transforms/compose.py#L313
https://github.com/Project-MONAI/MONAI/blob/8a70678baa976a01274f55837e339f2c2975dec7/monai/data/dataset.py#L808

Reply:
```is_random``` is an initial attempt to solve the problem. I've thought about it over the weekend and if I can avoid introducing ```is_random``` then I'll get rid of it. In particular, if we always iterate into Compose objects, we don't need a check to determine whether the Compose object contains only provably deterministic transforms.

```flatten``` has its own problems in that it isn't a complete mechanism and it produces a physically flattened set of transforms. I'm prepping another commit that iterates over transforms without flattening them and returns an index instead.